jdaviz/configs/imviz/wcs_utils.py

```diff
@@ -252,3 +253,25 @@ def draw_compass_mpl(image, orig_shape=None, wcs=None, show=True, zoom_limits=No
     plt.close()
 
     return base64.b64encode(buff.getvalue()).decode('utf-8')
+
+
+def data_within_gwcs_bounding_box(data, x, y):
+    """This is for internal use by Imviz coordinates transformation only."""
+    within_bounding_box = True
+    if hasattr(data.coords, '_orig_bounding_box'):
+        # then coords is a GWCS object and had its bounding box cleared
+        # by the Imviz parser
+        ints = data.coords._orig_bounding_box.intervals
+        if isinstance(ints[0].lower, u.Quantity):
+            bb_xmin = ints[0].lower.value
+            bb_xmax = ints[0].upper.value
+            bb_ymin = ints[1].lower.value
+            bb_ymax = ints[1].upper.value
+        else:  # pragma: no cover
+            bb_xmin = ints[0].lower
+            bb_xmax = ints[0].upper
+            bb_ymin = ints[1].lower
+            bb_ymax = ints[1].upper
+        if not (bb_xmin <= x <= bb_xmax and bb_ymin <= y <= bb_ymax):

```

Comment:
is the bounding box meant to be inclusive (should a pixel on the bounding box interval be considered valid or invalid for extrapolation)?

Reply:
https://gwcs.readthedocs.io/en/latest/gwcs/using_wcs.html#using-bounding-box isn't very helpful but since it is modeling, maybe @williamjamieson knows 🙏

Reply:
This function is what is used to determine if points are outside the the bounding box on a given interval: https://github.com/astropy/astropy/blob/7524bd185df2cc931fe2e374742a41cb768efbbf/astropy/modeling/bounding_box.py#L117-L130. It tests if it is strictly less than the lower bound and/or strictly greater than the upper bound. Therefore, each interval of the bounding box is a closed interval (i.e. the endpoints are included).

Reply:
Thanks!  @pllim - should we switch this to be `if ints[0].outside(x) or ints[1].outside(y)` then, or will that still have the quantities issue?  Otherwise, I think the logic here is at least consistent with that and so doesn't need to be changed.

Reply:
As a user of BoundingBox intervals. Would it be useful to also have an "inside" function as well (outside is what was needed for the astropy implementation but "inside" was not).

Reply:
> Thanks! @pllim - should we switch this to be `if ints[0].outside(x) or ints[1].outside(y)` then, or will that still have the quantities issue? 

I don't think I actually tested if outside will work with `Quantity` values. Let me know if it does.

Reply:
@WilliamJamieson - it might be useful to have it at the `BoundingBox` level as well (to pass x,y coordinates and know whether that falls inside or outside instead of having to know which interval is which and check them separately)!

Reply:
@kecnry would a more refined version of https://github.com/astropy/astropy/blob/7524bd185df2cc931fe2e374742a41cb768efbbf/astropy/modeling/bounding_box.py#L854-L888 work for you?  That private method returns an array shaped to the broadcasted common shape of x and y of booleans indicating which parts are inside and outside (it also returns if all the points will be outside). I can wrap it to be slightly more user friendly.

Reply:
Is that the same as https://gwcs.readthedocs.io/en/latest/api/gwcs.wcs.WCS.html#gwcs.wcs.WCS.in_image ?

Reply:
@kecnry , so is the code ok as-is or is there some action item here?

Reply:
the current code is correct, as far as I can tell.  We could switch to using the methods from bounding box, but I don't think its necessary (unless things get more complicated down the road with rotation, etc).